### PR TITLE
edit SaltStack license info so that GitHub recognizes it

### PR DESCRIPTION
**Pull request by eirinikos**
Hello! 👋

(CCing @dankohn who has requested this work so that the license will appear correctly in
https://landscape.cncf.io/selected=salt-stack)

### What does this PR do?
GitHub uses a library called Licensee to identify a project's license
type. It shows this information in the status bar and via the API if it
can unambiguously identify the license.

This commit modifies a few of SaltStack's docs so that Licensee is able
to recognize the repository's license type. LICENSE has been updated to
contain the full text of the Apache license. COPYING has been removed;
the info regarding external module licensing is now stored at the bottom
of LICENSE.

This commit also adds a "License" section to the README to clarify the
licensing for SaltStack as well as for external modules.

The update to LICENSE and the consolidation of COPYING with LICENSE
allow Licensee to successfully identify the license type of SaltStack as
Apache.

### What issues does this PR fix or reference?
There is no previously existing GitHub issue for this PR.

### Previous Behavior
Here is the output that I get when I run the Licensee command-line tool on SaltStack's remote repo:

```
$ licensee detect https://github.com/saltstack/salt
License:        Other
Matched files:  LICENSE, COPYING
LICENSE:
  Content hash:  570ad566bfcf2ff49dac56fcbfb5f7ee0529773e
  License:       Other
  Closest licenses:
    Unlicense similarity:  34.64%
    Zlib similarity:       32.17%
    MIT similarity:        31.45%
COPYING:
  Content hash:  13f75f7115f511118548777d1b318c8b11849170
  License:       Other
  Closest licenses:
    BSD-3-Clause-Clear similarity:  58.20%
    BSD-3-Clause similarity:        56.37%
    BSD-2-Clause similarity:        56.09%
```

### New Behavior
And here is the output that I get when I run Licensee on the `update-license` branch of my local SaltStack clone:

```
$ licensee detect ../salt/
License:        Apache License 2.0
Matched files:  LICENSE
LICENSE:
  Content hash:  4f3dee90d3ff02d566b067c25982c2bfd7e360c1
  Confidence:    100.00%
  Matcher:       Licensee::Matchers::Exact
  License:       Apache License 2.0
```

### Tests written?
No

### Commits signed with GPG?
No

### CODE REVIEW

File: README.rst
Comment (cachedout) [inline README.rst]: IMHO, this should explicitly say that it's the `2.0` version of the Apache license.

Comment (eirinikos) [reply README.rst]: Makes sense to me!

File: LICENSE
Comment (cachedout) [inline LICENSE]: If this gets pulled it needs to be re-added, probably down in L189.

**Review by cachedout — CHANGES_REQUESTED** [review]

File: LICENSE
Comment (cachedout) [reply LICENSE]: I have to defer to @thatch45 here. I suspect that moving this to the README is fine but I'm not completely sure.

Comment (thatch45) [reply LICENSE]: sorry for the delay, lets change it to SaltStack Team, lets not risk making this a legal event :)

Comment (rallytime) [reply LICENSE]: @eirinikos Did you see the reply above?

**Review by cachedout — CHANGES_REQUESTED** [review]
@eirinikos Bump. There is some review feedback to be addressed before we can merge this.

**Comment by eirinikos** [comment]
@thatch45 @rallytime @cachedout Thanks for the follow-up and sorry for the late reply. I'll push an update as soon as I can.

**Comment by dwoz** [comment]
@dmurphy18 Is any of this used by our packages?

**Comment by eirinikos** [comment]
Hi @cachedout - I just updated my commit, the changes should be good to go. 👍

**Review by cachedout — APPROVED** [review]

**Comment by rallytime** [comment]
@thatch45 Can you take a look here one more time?

**Review by thatch45 — APPROVED** [review]

**Pull request merged**
